src/base/bittorrent/torrentinfo.cpp

```diff
@@ -76,13 +76,31 @@ TorrentInfo TorrentInfo::load(const QByteArray &data, QString *error) noexcept
 
 TorrentInfo TorrentInfo::loadFromFile(const QString &path, QString *error) noexcept
 {
-    libt::error_code ec;
-    TorrentInfo info(NativePtr(new libt::torrent_info(Utils::Fs::toNativePath(path).toStdString(), ec)));
+    if (error)
+        error->clear();
+
+    QFile file {path};
+    file.open(QIODevice::ReadOnly);
+    const QByteArray data = file.read(100 * 1024 * 1024);  // 100 MB
+    file.close();
+
+    // 2-step construction to overcome default limits of `depth_limit` & `token_limit` which are
+    // used in `torrent_info()` constructor
+    const int depthLimit = 100;
+    const int tokenLimit = 10000000;
+    libt::bdecode_node node;
+    libt::error_code decodeEc;
+    bdecode(data.constData(), (data.constData() + data.size()), node, decodeEc
+        , nullptr, depthLimit, tokenLimit);
+
+    libt::error_code infoEc;
+    TorrentInfo info(NativePtr(new libt::torrent_info(node, infoEc)));
```

Comment:
I wouldn't try create TorrentInfo not checking for bdecode errors first.

Reply:
pardon me, fixed now.